requests/models.py

```diff
@@ -796,7 +800,24 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             if delimiter:
                 lines = chunk.split(delimiter)
             else:
+                # Python splitlines() supports the universal newline (PEP 278).
+                # That means, '\r', '\n', and '\r\n' are all treated as end of
+                # line. If the last chunk ends with '\r', and the current chunk
+                # starts with '\n', they should be merged and treated as only
+                # *one* new line separator '\r\n' by splitlines().
+                # This rule only applies when splitlines() is used.
+                if last_chunk_ends_with_cr and chunk.startswith(line_feed):
```

Comment:
Now that I understand what concerns @Lukasa has, I agree. I think this could be better written as:

```py
chunk_startswith_line_feed = last_chunk_ends_with_cr and chunk.startswith(line_feed)
last_chunk_ends_with_cr = chunk.endswith(carriage_return)
if chunk_startswith_line_feed:
    chunk = chunk[1:]
    if not chunk:
        continue
lines = chunk.splitlines()
```

This properly *explains* the condition on this line with a descriptive variable name. It avoids the `old_last_chunk_ends_with_cr` variable and it keeps things concise and easy to reason about.